tests/unit/modules/test_mysql.py

```diff
@@ -282,6 +282,23 @@ def test_free_slave(self):
     def test_query(self):
         self._test_call(mysql.query, 'SELECT * FROM testdb', 'testdb', 'SELECT * FROM testdb')
 
+    def test_query_error(self):
+        def mock_execute(cur, query):
+            '''
+            Mock of info method
+            '''
+            raise MySQLdb.OperationalError(9999, 'Something Went Wrong')
+
+        connect_mock = MagicMock()
+        with patch.object(mysql, '_connect', connect_mock):
+            with patch.dict(mysql.__salt__, {'config.option': MagicMock()}):
+                with patch.object(mysql, '_execute') as execute_mock:
+                    execute_mock.side_effect = mock_execute
```

Comment:
You can actually setup the side_effect right in the init for `MagicMock()`:

```python
MagicMock(side_effect=MySQLdb.OperationalError(9999, 'Something Went Wrong'))
```

This would remove the need to A) write a separate function, and B) set the side_effect manually after the MagicMock is initialized.

It'll work perfectly fine as-is, so I'm approving the PR, but just something to keep in mind.

Reply:
Updating to the more concise approach. Thanks!